test/python/circuit/test_circuit_operations.py

```diff
@@ -852,6 +852,32 @@ def test_measure_all_repetition(self):
         self.assertEqual(len(circuit.cregs[0]), 2)  # Both length 2
         self.assertEqual(len(circuit.cregs[1]), 2)
 
+    def test_measure_all_with_multiple_regs_creation(self):
+        """Test measure_all in a circuit where the method is called
+        multiple times consecutively, and checks that a register of
+        a different name is created on each call."""
+
+        circuit = QuantumCircuit(1)
+
+        # First call should create a new register
+        circuit.measure_all()
+        self.assertEqual(len(circuit.cregs), 1)  # One creg
+        self.assertEqual(len(circuit.cregs[0]), 1)  # Of length 1
+
+        # Second call should also create a new register
+        circuit.measure_all()
+        self.assertEqual(len(circuit.cregs), 2)  # Now two cregs
+        self.assertTrue(all(circuit.cregs), lambda reg: len(reg) == 1)  # All of length 1
```

Comment:
I think you probably want: 

```suggestion
        self.assertTrue(all(len(reg) == 1 for reg in circuit.cregs))  # All of length 1
```